tests/test_x509.py

```diff
@@ -644,6 +644,127 @@ def test_public_bytes_match(self, request_path, loader_func, encoding,
         assert serialized == request_bytes
 
 
+@pytest.mark.requires_backend_interface(interface=RSABackend)
+@pytest.mark.requires_backend_interface(interface=X509Backend)
+class TestCertificateSigningRequestBuilder(object):
+    def test_sign_invalid_hash_algorithm(self, backend):
+        private_key = rsa.generate_private_key(
+            public_exponent=65537,
+            key_size=2048,
+            backend=backend,
+        )
+        builder = x509.CertificateSigningRequestBuilder()
+        with pytest.raises(TypeError):
+            builder.sign(backend, private_key, 'NotAHash')
+
+    def test_build_ca_request(self, backend):
+        private_key = rsa.generate_private_key(
+            public_exponent=65537,
+            key_size=2048,
+            backend=backend,
+        )
+
+        request = x509.CertificateSigningRequestBuilder().subject_name(
+            x509.Name([
+                x509.NameAttribute(x509.OID_COUNTRY_NAME, 'US'),
+                x509.NameAttribute(x509.OID_STATE_OR_PROVINCE_NAME, 'Texas'),
+                x509.NameAttribute(x509.OID_LOCALITY_NAME, 'Austin'),
+                x509.NameAttribute(x509.OID_ORGANIZATION_NAME, 'PyCA'),
+                x509.NameAttribute(x509.OID_COMMON_NAME, 'cryptography.io'),
+            ])
+        ).add_extension(
+            x509.BasicConstraints(True, 2), critical=True
```

Comment:
Actually fixed in https://github.com/sigmavirus24/cryptography/commit/eece6d4ce8e50ed0f4869b062a5c77679e797b34